magma/circuit.py

```diff
@@ -758,6 +763,15 @@ def is_definition(self):
     def instances(self):
         return self._context_.placer.instances()
 
+    def get_instance(self, name: str) -> 'AnonymousCircuitType':
+        instances = self.instances
+        try:
+            return only(filter(lambda i: i.name == name, instances))
```

Comment:
Could refactor this into a helper function to share code with https://github.com/phanrahan/magma/pull/1263/files#diff-889a16f7db04ed3ceb8ae4a89693b806f06d3e0c3894022ed50999a1831ce874R752

Reply:
Good idea - but I kind of like that we can use these functions in an almost functional pattern, and it's a single line. Think refactoring would increase the total number of lines :).